### PR TITLE
Add note about assembly headers

### PR DESCRIPTION
**Pull request by bobfuru**
Updating guidance about "Next steps" and "Prerequisites" formatting, as described in https://github.com/openshift/openshift-docs/pull/25032 and specifically in Andrea's comment: https://github.com/openshift/openshift-docs/pull/25032#issuecomment-681087488

The purpose is to give guidance that `.` should be used in modules to avoid rendering of a TOC inside the module. I did not update the guidance for "Related information" (or change it to "Additional information", although this seems to be more common) because the guidelines don't show an example of using `==` for this. But maybe it should?

@openshift/team-documentation @vikram-redhat @bergerhoffer PTAL

### CODE REVIEW

**Review by lbarbeevargas — COMMENTED** [review]
The new notes LGTM. 

If "Related information" can be used in assemblies as well as modules, my vote is to add a similar `== Related information` example here in addition to a note about using it in an assembly vs module.

Keeping the `Peer Review Needed` label on the PR pending Andrea's review.

**Review by bergerhoffer — COMMENTED** [review]
So your notes that you added are good, but I don't think that the existing example presented makes sense at all. It looks to be a combo of an assembly and a module. Because it has a == Prerequisites, then a .Procedure, and then == Next steps (which are usually normally only used in assemblies since we currently can't xref in modules.

Also "Related information" is not one of the approved headings per mod docs. Only "Next steps" or "Additional resources" are.

So here's what I'd suggest:

1. Remove this example code block completely and instead link to the template for a procedure module from the mod docs repo. That way it's up to date always with the guidelines
2. After removing the example, add in a note about using . syntax for Prereqs/Next steps, etc.
3. Add in a section above "Writing concepts" about "Writing assemblies", with the link to the mod docs assembly template.
4. In the new assembly section, add in a note about using == syntax for Prereqs/Next steps, etc.

If you want to make the update in this PR that's cool, or I'd be happy to work on it if you want to merge this PR in first.

File: contributing_to_docs/doc_guidelines.adoc [hunk 1]
```diff
@@ -128,62 +128,41 @@ ID. For example:
...
-link:https://redhat-documentation.github.io/modular-docs/[Red Hat modular docs reference guide].
+== Writing assemblies
+An _assembly_ is a collection of modules that describes how to accomplish a user story.
 
-== Writing tasks
-A _task_ contains the steps that users follow to complete a process. Tasks
-contain ordered steps and explicit commands. In most cases, create your tasks
-as individual modules and include them in appropriate assemblies.
+For more information about forming assemblies, see the
+link:https://redhat-documentation.github.io/modular-docs/[Red Hat modular docs reference guide] and the link:https://github.com/redhat-documentation/modular-docs/blob/master/modular-docs-manual/files/TEMPLATE_ASSEMBLY_a-collection-of-modules.adoc[assembly template].
```
Comment (bergerhoffer) [inline contributing_to_docs/doc_guidelines.adoc:135]: I would link to the raw version of the template: https://raw.githubusercontent.com/redhat-documentation/modular-docs/master/modular-docs-manual/files/TEMPLATE_ASSEMBLY_a-collection-of-modules.adoc

File: contributing_to_docs/doc_guidelines.adoc [hunk 2]
```diff
@@ -128,62 +128,41 @@ ID. For example:
...
 
-== Next steps
+Avoid using gerunds in concept titles. "About <concept>"
+is a common concept module title.
 
-You can explain any other tasks that MUST be completed after this task. You can
-skip this if there are none. Do not include xrefs. If the next steps are closely
-related to the task, you might be able to include their modules in the assembly.
+For more information about creating concept modules, see the
+link:https://redhat-documentation.github.io/modular-docs/[Red Hat modular docs reference guide] and the link:https://github.com/redhat-documentation/modular-docs/blob/master/modular-docs-manual/files/TEMPLATE_CONCEPT_concept-explanation.adoc[concept template].
```
Comment (bergerhoffer) [inline contributing_to_docs/doc_guidelines.adoc:152]: Same, link to the raw version

File: contributing_to_docs/doc_guidelines.adoc [hunk 3]
```diff
@@ -128,62 +128,41 @@ ID. For example:
...
+== Writing tasks
+A _task_ contains the steps that users follow to complete a process. A task, also known as a _procedure_, contains ordered steps and explicit commands. In most cases, create your tasks as individual modules and include them in appropriate assemblies.
 
-* A bulleted list of links to related information about this task. Skip if there isn't any related information.
-----
+Use a verb in the task title, such as "Creating".
 
 For more information about writing tasks, see the
-link:https://redhat-documentation.github.io/modular-docs/[Red Hat modular docs reference guide].
+link:https://redhat-documentation.github.io/modular-docs/[Red Hat modular docs reference guide] and the link:https://github.com/redhat-documentation/modular-docs/blob/master/modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc[procedure template].
```
Comment (bergerhoffer) [inline contributing_to_docs/doc_guidelines.adoc:160]: Same, link to the raw version

File: contributing_to_docs/doc_guidelines.adoc (hunk 1 above)
Comment (bergerhoffer) [reply contributing_to_docs/doc_guidelines.adoc:135]: Also for the first link, maybe go straight to the assembly section? https://redhat-documentation.github.io/modular-docs/#forming-assemblies

File: contributing_to_docs/doc_guidelines.adoc (hunk 2 above)
Comment (bergerhoffer) [reply contributing_to_docs/doc_guidelines.adoc:152]: For the first link, maybe link directly to the concept section? https://redhat-documentation.github.io/modular-docs/#creating-concept-modules

File: contributing_to_docs/doc_guidelines.adoc (hunk 3 above)
Comment (bergerhoffer) [reply contributing_to_docs/doc_guidelines.adoc:160]: For the first link, link to the procedure section? https://redhat-documentation.github.io/modular-docs/#creating-procedure-modules

File: contributing_to_docs/doc_guidelines.adoc [hunk 4]
```diff
@@ -128,62 +128,41 @@ ID. For example:
...
+is a common concept module title.
 
-You can explain any other tasks that MUST be completed after this task. You can
-skip this if there are none. Do not include xrefs. If the next steps are closely
-related to the task, you might be able to include their modules in the assembly.
+For more information about creating concept modules, see the
+link:https://redhat-documentation.github.io/modular-docs/[Red Hat modular docs reference guide] and the link:https://github.com/redhat-documentation/modular-docs/blob/master/modular-docs-manual/files/TEMPLATE_CONCEPT_concept-explanation.adoc[concept template].
 
-.Related information
+== Writing tasks
```
Comment (bergerhoffer) [inline contributing_to_docs/doc_guidelines.adoc:154]: Suggest replacing "task" with "procedure" throughout this section, since that's the terminology we use.

File: contributing_to_docs/doc_guidelines.adoc [hunk 5]
```diff
@@ -128,62 +128,41 @@ ID. For example:
...
-----
+Use a verb in the task title, such as "Creating".
 
 For more information about writing tasks, see the
-link:https://redhat-documentation.github.io/modular-docs/[Red Hat modular docs reference guide].
+link:https://redhat-documentation.github.io/modular-docs/[Red Hat modular docs reference guide] and the link:https://github.com/redhat-documentation/modular-docs/blob/master/modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc[procedure template].
+
+[NOTE]
+====
+If you include `Prerequistes`, `Next steps`, or `Additional resources` headings in your procedure, use only `==` formatting in assembly files. For module files, use `.Prerequisites`, `.Next steps`, or `.Additional information` syntax to avoid TOC formatting within the module. Do not include xrefs in modules.
```
Comment (bergerhoffer) [inline contributing_to_docs/doc_guidelines.adoc:164]: Suggest making this more targeted to modules. I almost read this first as it should use == since that is what is listed first. Maybe simplify to something like:

```
"If the sections are applicable to your module, use the following syntax for these standard headings: ".Prerequisites", ".Next steps", ".Additional resources". Do not use "==" headings for these sections in a module. Remember that modules cannot include any xrefs.
```
^ I don't love my wording attempt, but you get the idea - to include the part for modules first before saying what NOT to do.

File: contributing_to_docs/doc_guidelines.adoc [hunk 6]
```diff
@@ -128,62 +128,41 @@ ID. For example:
...
-contain ordered steps and explicit commands. In most cases, create your tasks
-as individual modules and include them in appropriate assemblies.
+For more information about forming assemblies, see the
+link:https://redhat-documentation.github.io/modular-docs/[Red Hat modular docs reference guide] and the link:https://github.com/redhat-documentation/modular-docs/blob/master/modular-docs-manual/files/TEMPLATE_ASSEMBLY_a-collection-of-modules.adoc[assembly template].
 
-[IMPORTANT]
+[NOTE]
 ====
-Use a verb in the task title (for example, Creating).
+For `Prerequistes`, `Next steps`, and `Additional resources` headings, use only `==` formatting in assembly files. For module files, use `.Prerequisites`, `.Next steps`, or `.Additional information` syntax to avoid TOC formatting within the module. Do not include xrefs in modules.
```
Comment (bergerhoffer) [inline contributing_to_docs/doc_guidelines.adoc:139]: * I would reword the "For module files" sentence to specifically say to NOT use the .Title syntax for these headings in assembly files.
* I don't think the "don't xref in modules" part is applicable for this assembly section.

File: contributing_to_docs/doc_guidelines.adoc [hunk 7]
```diff
@@ -128,62 +128,41 @@ ID. For example:
...
+For more information about creating concept modules, see the
+link:https://redhat-documentation.github.io/modular-docs/[Red Hat modular docs reference guide] and the link:https://github.com/redhat-documentation/modular-docs/blob/master/modular-docs-manual/files/TEMPLATE_CONCEPT_concept-explanation.adoc[concept template].
 
-.Related information
+== Writing tasks
+A _task_ contains the steps that users follow to complete a process. A task, also known as a _procedure_, contains ordered steps and explicit commands. In most cases, create your tasks as individual modules and include them in appropriate assemblies.
 
-* A bulleted list of links to related information about this task. Skip if there isn't any related information.
-----
+Use a verb in the task title, such as "Creating".
```
Comment (bergerhoffer) [inline contributing_to_docs/doc_guidelines.adoc:157]: I would add "gerund" in there

**Review by bergerhoffer — COMMENTED** [review]
A few suggestions :)

File: contributing_to_docs/doc_guidelines.adoc (hunk 5 above)
Comment (bobfuru) [reply contributing_to_docs/doc_guidelines.adoc]: Thanks, @bergerhoffer! I tried to reword so that it explains how to format if it is a module, while still leaving the possibility that it might be an assembly. But didn't want to over-complicate it too much by adding something like, "assembly files can use `==` formatting. Let me know if it doesn't seem clear enough.

File: contributing_to_docs/doc_guidelines.adoc [hunk 8]
```diff
@@ -128,62 +128,41 @@ ID. For example:
...
+== Writing procedures
+A _procedure_ contains the steps that users follow to complete a process or task. Procedures contain ordered steps and explicit commands. In most cases, create your procedures as individual modules and include them in appropriate assemblies.
 
-.Related information
+Use a gerund in the procedure title, such as "Creating".
 
-* A bulleted list of links to related information about this task. Skip if there isn't any related information.
-----
+For more information about writing procedures, see the
+link:https://redhat-documentation.github.io/modular-docs/#creating-procedure-modules/[Red Hat modular docs reference guide] and the link:https://raw.githubusercontent.com/redhat-documentation/modular-docs/master/modular-docs-manual/files/TEMPLATE_PROCEDURE_doing-one-procedure.adoc[procedure template].
```
Comment (bergerhoffer) [inline contributing_to_docs/doc_guidelines.adoc:160]: The link doesn't work with the ending slash

**Review by bergerhoffer — COMMENTED** [review]
One last thing

File: contributing_to_docs/doc_guidelines.adoc (hunk 5 above)
Comment (bergerhoffer) [reply contributing_to_docs/doc_guidelines.adoc]: Ah, I think I see what you're getting at, that procedures could be directly in an assembly? I would think that all procedures should really be in a module, but perhaps there is a use case for being directly in an assembly. So that's fine to keep as you have it.

Comment (bobfuru) [reply contributing_to_docs/doc_guidelines.adoc]: I feel like I've seen a proc as an assembly but not sure where, and it would certainly be an exception. I've tightened up the wording a bit to favor procedures as modules.

File: contributing_to_docs/doc_guidelines.adoc (hunk 8 above)
Comment (bobfuru) [reply contributing_to_docs/doc_guidelines.adoc]: resolved

File: contributing_to_docs/doc_guidelines.adoc (hunk 5 above)
Comment (adellape) [reply contributing_to_docs/doc_guidelines.adoc]: :rotating_light: Ancient thread revival thanks to the BLAME button :rotating_light: 

The way this is written still seems to be confusing people into thinking you can't use `.Additional resources` in a module (which, because they often include an `xref`, a lot of times get actually attached right after the module's include at the assembly level, but are not intended to be at the assembly's TOC level). The NOTE further down in the Writing procedures section is more clear on this, but maybe they're just stopping with a CTRL+F on "Additional r..."

Also it still says "Additional information", which I assume is meant to meant "Additional resources" (since the former doesn't appear anywhere in the [mod docs guide](https://redhat-documentation.github.io/modular-docs/)).

Happy to adjust these things myself, but just wanted to make sure I wasn't misunderstanding the original intent here.

Comment (bobfuru) [reply contributing_to_docs/doc_guidelines.adoc]: I agree with @adellape that this needs work. Addressed in https://github.com/openshift/openshift-docs/pull/30125 - thanks!
